src/components/calcite-input/calcite-input.tsx

```diff
@@ -278,11 +278,19 @@ export class CalciteInput {
   componentDidLoad(): void {
     this.slottedActionEl = this.el.querySelector("[slot=input-action]");
     this.setDisabledAction();
-    if (this.shouldFormatNumberByLocale()) {
+    if (this.type === "number") {
       this.childEl.style.cssText = hiddenInputStyle;
     }
   }
 
+  componentShouldUpdate(newValue: string, oldValue: string, property: string): boolean {
+    if (this.type === "number" && property === "value" && newValue && !isValidNumber(newValue)) {
+      this.value = oldValue;
+      return false;
```

Comment:
Nitpick: Would it be possible to store the conditional expression into an expressive var and return that instead? Totally optional.